### PR TITLE
Remove `command.bash` and deprecate `command-hook-asdf.bash`.

### PR DESCRIPTION
**Pull request by vic**
Remove command.bash. 

The old `asdf direnv` command is not needed anymore since it was only a wrapper for direnv itself
and a very hackish way to invoke what is now `asdf direnv envrc` command.

Previously we instructed users how to setup their direnv integration.
The old `asdf direnv hook asdf` exported a `use_asdf` function.
That command is now deprecated and will show an error telling the
user to remove the asdf integration from `~/.config/direnv/direnvrc`
and run the new `setup` command.

Now, since we already are automating things, the `setup` command will
have ownership of its own `use_asdf.sh` file, where it defines the `use_asdf` 
function for direnv to use.  We simply define it as `source_env $(asdf direnv envrc)`.

One advantage is that direnv will automatically watch the cahed envrc
produced by `asdf direnv envrc`. Another one is that people can
directly call this same command which will print the path to their
cached environment, eg for inspecting or removing it.

Modified test environment to do `asdf direnv setup` on all tests.
Also now they use `asdf direnv local` to activate the local environment
just as users would.

### CODE REVIEW

**Comment by vic** [comment]
This one is also waiting for #129 since setup command will change and we use it to prepate test environment on this branch.

File: lib/commands/command-hook-asdf.bash [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
+# TODO: remove once we release 0.4.0
+cat <<-'EOF' >&2
+Looks like you are using the 'asdf direnv hook asdf' command which was removed in asdf-direnv 0.3.0.
 
-if [ "1" == "$DIRENV_IN_ENVRC" ] && [ "$0" == "${BASH_SOURCE[0]}" ]; then
-  echo "$0"
-fi
+Remove the line that reads 'asdf direnv hook asdf' from your rc-file: ~/.config/direnv/direnvrc
+
+And run `asdf direnv setup --help` for using the new setup.
```
Comment (jfly) [inline lib/commands/command-hook-asdf.bash:9]: Nit: "for using" -> "for instructions to perform"?

File: lib/setup-lib.bash [hunk 2]
```diff
@@ -244,8 +246,8 @@ function local_command() {
     maybe_run_cmd asdf plugin-add "$plugin"
     run_cmd asdf install "$plugin" "$version"
     run_cmd asdf local "$plugin" "$version"
-
-    printf "use asdf\n\0" | grep_or_add ".envrc"
-    run_cmd direnv allow
   done
+
+  printf "use asdf\n\0" | grep_or_add ".envrc"
+  run_cmd direnv allow
```
Comment (jfly) [inline lib/setup-lib.bash:252]: Oops! Good call on moving this out of the loop.

File: lib/commands/command-hook-asdf.bash [hunk 3]
```diff
@@ -1,14 +1,11 @@
 #!/usr/bin/env bash
 
-use_asdf() {
-  if ! env_file="$(asdf direnv _asdf_cached_envrc "$@")"; then
-    log_error "Error generating asdf cached envrc"
```
Comment (jfly) [inline lib/commands/command-hook-asdf.bash:5]: What happens now when we fail to generate an envrc file? Do we still get a nice error message?

**Review by jfly — APPROVED** [review]
This is awesome! I particularly like that our tests are now dogfooding our new setup script.

**Pull request merged**